### PR TITLE
Feature: Random mentor suggestions endpoint 

### PR DESCRIPTION
**Pull request by ulloa**
## Description

Added an endpoint for random mentor suggestions. This is so that we can start hooking up data between the web api and data science api.

[Trello Card 159](https://trello.com/c/h4lW4rIU)

https://www.loom.com/share/065e1fd52536410a97616eebddba6440

- [x] New feature

## Checklist:

- [x] My code follows PEP8 style guide
- [x] I have removed unnecessary print statements from my code
- [x] I have made corresponding changes to the documentation if necessary
- [x] My changes generate no errors
- [x] No commented-out code
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

**Comment by RylandGomez** [comment]
Just a thought, perhaps it would be better to assign the mentors variable within the API route? This would avoid having to import the database within the ML interface and improve encapsulation, in my mind. I'd like to discuss if possible before further reviewing.

**Comment by ulloa** [comment]
I'm making that change right now @RylandGomez

**Review by RylandGomez — CHANGES_REQUESTED** [review]
First off, thank you for adding __pycache__/ into the .gitignore file haha
Beyond that, the functionality for the code seems sound. My issue lies in the implementation. Specifically, the Matcher class currently seems to break encapsulation. I outlined my thoughts on how this could potentially be addressed with line comments.
If I'm wrong on any of this, feel free to call me out and discuss!

File: app/model.py [hunk 1]
```diff
@@ -3,8 +3,41 @@
...
 """
 from sklearn.metrics.pairwise import cosine_similarity
+from app.data import MongoDB
 import pandas as pd
+import numpy as np
 
 
-class Matcher:
-    pass
+db = MongoDB("UnderdogDevs")
```
[the remark this answers is not in the document]
Comment (ulloa) [reply app/model.py]: I think this one I fixed already, with exception to the when run as main.

File: app/api.py [hunk 2]
```diff
@@ -70,6 +70,17 @@ async def search(collection_name: str, user_search: str):
     return {"result": API.db.search(collection_name, user_search)}
 
 
+@API.post("/random_mentors")
+async def random_mentors(num_recommendations: int):
+    """ accepts N as the number of recommendations to return """
+    matcher = Matcher()
```
[the remark this answers is not in the document]
Comment (ulloa) [reply app/api.py]: I'll look into this. Anyone else sure about it?

File: app/model.py [hunk 3]
```diff
@@ -3,8 +3,41 @@
...
+    def random_recommendations(self, N=5):
+        """
+        Reads all mentors in db, then returns a list of length N random mentors
+
+        Parameters
+        ----------
+            N : int
+                number of recommendations to return
+        """
+        mentors = db.read("Mentors", None)
```
[the remark this answers is not in the document]
Comment (ulloa) [reply app/model.py]: With the db as an argument of the Matcher class now, can you look see if it should still be changed? 

If I go into the api.py, and change pass the following as an argument to `Matcher(API.db.dataframe('mentors'))` the `matcher.random_recommendations`, that could also work. 

Also, now I'm considering changing the name from random_recommendations to a verb.

**Review by RylandGomez — CHANGES_REQUESTED** [review]
Line comments go into detail, but tl;dr the assert in class Matcher doesn't seem appropriate, and encapsulation issue in model.py persists.

File: app/api.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (RylandGomez) [reply app/api.py]: iirc in office hours, Robert mentioned something about instantiating classes one level higher than where they are going to be used. My memory's a little foggy on the specifics, though, which is why I'm not so confident in raising this as an error.

[the remark this answers is not in the document]
Comment (BrokenShell) [reply app/api.py]: This is a great conversation! We'll definitely talk about this more in office hours.

**Comment by ulloa** [comment]
I'm closing this for now. I believe #37 is the better direction for this.

**Pull request closed without merging**